src/executor/mod.rs

```diff
@@ -187,9 +188,14 @@ impl EvmExecutor {
                     }
                 }
             }
-            if (acc.is_empty() && !info.is_empty()) || acc.code_hash.0 != info.code_hash.0 {
-                acc_data.poseidon_code_hash = H256::from(info.code_hash.0);
-                acc_data.keccak_code_hash = H256::from(info.keccak_code_hash.0);
+            if (acc.is_empty() && !info.is_empty()) || acc.keccak_code_hash.0 != info.code_hash.0 {
```

Comment:
Since there are several conditions to trigger this, inline documentation would help.